sktime/forecasting/base/_fh.py

```diff
@@ -212,9 +257,11 @@ def _new(
         values : pd.Index, pd.TimedeltaIndex, np.array, list, pd.Timedelta, or int
             Values of forecasting horizon.
         is_relative : bool, default=same as self.is_relative
-        - If None, determined automatically: same as self.is_relative
-        - If True, values are relative to end of training series.
-        - If False, values are absolute.
+            - If None, determined automatically: same as self.is_relative
+            - If True, values are relative to end of training series.
+            - If False, values are absolute.
+        freq : str, optional (default=None)
```

Comment:
did you consider making this polymorphic, as discussed in the STEP?

I.e., allowing multiple typers of input for `freq`, including `pandas` data containers or indices? If those are passed, then `infer_freq` will unwrap. So you can do `_check_fh(fh, y)` instead of `_check_fh(fh, index)`

Reply:
(also the docstring should reflect that)

Reply:
I think the docstrings are now not accurate anymore, could you kindly update to what is actual?

Reply:
Which docstrings do you have in mind? I briefly went through those that are affected by this PR and they look ok to me.